src/tests/conformance/handshake.rs

```diff
@@ -72,3 +75,90 @@ async fn handshake_initiator_side() {
 
     node.stop().await;
 }
+
+#[tokio::test]
+async fn conformance_004() {
```

Comment:
I'm not sure what to name these tests. I'd prefer names that indicate what they're testing, but some prefer being able to trace the spec to the test.

Reply:
How about naming them nicely whilst commenting the spec ID just under it? At the very least, this should make it easy for us to switch later if necessary.